src/main/resources/templates/mdBenefitsFlow/mailingAddress.html

```diff
@@ -41,5 +41,38 @@
   </section>
 </div>
 <th:block th:replace="~{fragments/footer :: footer}"/>
+<script th:inline="javascript">
+  var inputData = [[${inputData}]];
+
+  // Function to update address fields based on the checkbox
+  function updateAddressFields(onPageLoad = false) {
+    var checkbox = document.getElementById("sameAsHomeAddress-true");
+    if (checkbox && checkbox.checked) {
+      // Copy values from home address to mailing address
+      document.getElementById("mailingAddressStreetAddress1").value = inputData['homeAddressStreetAddress1'] || '';
+      document.getElementById("mailingAddressStreetAddress2").value = inputData['homeAddressStreetAddress2'] || '';
+      document.getElementById("mailingAddressCity").value = inputData['homeAddressCity'] || '';
+      document.getElementById("mailingAddressState").value = inputData['homeAddressState'] || '';
+      document.getElementById("mailingAddressZipCode").value = inputData['homeAddressZipCode'] || '';
+    } else {
+      if (!onPageLoad) {
+        // Clear mailing address fields if checkbox is unchecked
```

Comment:
I think you need to consider "onPageLoad" for the checkbox as well.

I am seeing my corrected mailing address  (was same as home address, so the checkbox is clicked) getting overwritten again by the home address, when I go back to the page. 

Scenario: I enter a home address and submit it; then on the mailing page I click that my mailing address is the same as my home address. On the next page I choose the corrected version and click continue.  Then I go back to the mailing address page and see the old (uncorrected) home address.  

The only time the home address should override the mailing is if I actively click on the box on that - not if it's set when the page loads.